pkg/capabilities/consensus/ocr3/transmitter.go

```diff
@@ -69,7 +69,10 @@ func (c *ContractTransmitter) Transmit(ctx context.Context, configDigest types.C
 		c.capability = cp.(capabilities.CallbackCapability)
 	}
 
-	_, err = c.capability.Execute(ctx, capabilities.CapabilityRequest{
+	// Execute synchronously and wait for a response: libocr's passed-in context is cancelled
+	// the moment we return, so without this we could hit a race condition where the context is cancelled
+	// before we've actually transmitted the result upstream.
+	_, err = capabilities.ExecuteSync(ctx, c.capability, capabilities.CapabilityRequest{
```

Comment:
I'm afraid this is still affected by ctx being cancelled (in fact I just tried in a local run and I'm getting a lot of "context deadline exceeded" + "timed out after 60.000000 seconds", which is not really the case - it times out immediately)

Reply:
Ignore the comment above. The fix is probably fine but I might've discovered an unrelated problem.

Reply:
OK, after dealing with some other unrelated problems I tried this fix and realized that:
1. OCR3 capability Execute() with methodSendResponse returns (nil,nil) on success and that doesn't play well with ExecuteSync(). gRPC server wraps it in a non-nil channel but ExecuteSync still waits for values.
2. I hacked passing a fake channel with an empty response in it and stuff started to work consistently!
3. I'm still seeing similar context-cancelled errors but they don't seem to prevent successful transmissions (maybe an artifact of not closing a channel in my hacky solution?).

Reply:
Yep, that's where I left it last night :) Will update the PR to ensure we return properly.

Side note: this type of thing is what I was referring to when talking about the awkwardness of using channels.